code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -35,7 +35,15 @@
 		user.visible_message("<span class='notice'>[user] finishes eating \the [src].</span>", \
 		"<span class='notice'>You finish eating \the [src].</span>")
 		score["foodeaten"]++ //For post-round score
-		user.drop_from_inventory(src) //Drop our item before we delete it
+
+		//Drop our item before we delete it, to clear any references of ourselves in people's hands or whatever.
+		if(loc == user)
+			user.drop_from_inventory(src)
+		else if(ismob(loc))
+			var/mob/holder = loc
+			holder.drop_from_inventory(src)
+		else //????
+			forceMove(get_turf(src))

```

Comment:
In all of these cases you want loc = null, which is what qdel should do anyway so...?